File: app/src/main/java/hr/caellian/notestream/gui/ActivityArtists.java
package hr.caellian.notestream.gui;

import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.NavigationView;
import android.widget.TableRow;
import android.widget.TextView;

import java.util.Map;

import hr.caellian.notestream.NoteStream;
import hr.caellian.notestream.R;
import hr.caellian.notestream.data.Playlist;
import hr.caellian.notestream.gui.fragments.FragmentCategoryTile;
import hr.caellian.notestream.util.Util;

/**
 * Created by tinsv on 22/07/2017.
 */

public class ActivityArtists extends TableActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(null);

        navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);

        ((TextView) findViewById(R.id.labelTableTitle)).setText(R.string.label_artists);

        FragmentManager fm = getFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();

        int counter = 0;
        TableRow row = addTableRow();
        for (Map.Entry<String, Playlist> artist : NoteStream.getInstance().library.artists.entrySet()) {
            FragmentCategoryTile fragment = FragmentCategoryTile.newInstance(
                    true,
                    R.drawable.ic_artist,
                    artist.getKey(),
                    artist.getValue());
            ft.add(row.getId(), fragment , Playlist.ARTIST_PREFIX + artist.getKey());
            counter = ++counter % 2;
            if (counter == 0) row = addTableRow();
        }
        ft.commit();
    }
}

File: app/src/main/java/hr/caellian/notestream/gui/ActivityArtists.kt
/*
 * Copyright (C) 2018 Tin Svagelj
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
 * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
 * persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
 * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
 * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package hr.caellian.notestream.gui

import android.os.Bundle
import android.widget.TextView
import hr.caellian.notestream.NoteStream
import hr.caellian.notestream.R
import hr.caellian.notestream.data.playlist.Playlist
import hr.caellian.notestream.gui.fragments.FragmentPlaylistTile
import hr.caellian.notestream.lib.Constants

class ActivityArtists : TableActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(null)

        navigationView = findViewById(R.id.nav_view)
        navigationView?.setNavigationItemSelectedListener(this)

        findViewById<TextView>(R.id.labelTableTitle).setText(R.string.label_artists)

        val fm = fragmentManager
        val ft = fm.beginTransaction()

        var counter = 0
        var row = addTableRow()
        for (artist in NoteStream.instance?.library?.artists?.entries
                ?: emptyList<MutableMap.MutableEntry<String, Playlist>>()) {
            val fragment = FragmentPlaylistTile.create(artist.value, R.drawable.ic_album, true)
            ft.add(row.id, fragment, Constants.PLAYLIST_ARTIST_PREFIX + artist.key.replace("", ""))
            counter = ++counter % 2
            if (counter == 0) row = addTableRow()
        }
        ft.commit()
    }
}
